nautobot/ipam/models.py

```diff
@@ -310,11 +356,19 @@ def clean(self):
         if not self.name:
             self.name = self.vrf.name
 
-        # A VRF must belong to a Device *or* to a VirtualMachine.
+        # A VRF must belong to a Device *or* to a VirtualMachine *or* to a Virtual Device Context.
         if all([self.device, self.virtual_machine]):
             raise ValidationError("A VRF cannot be associated with both a device and a virtual machine.")
-        if not any([self.device, self.virtual_machine]):
-            raise ValidationError("A VRF must be associated with either a device or a virtual machine.")
+        if all([self.device, self.virtual_device_context]):
+            raise ValidationError("A VRF cannot be associated with both a device and a virtual device context.")
+        if all([self.virtual_machine, self.virtual_device_context]):
+            raise ValidationError(
+                "A VRF cannot be associated with both a virtual machine and a virtual device context."
+            )
+        if not any([self.device, self.virtual_machine, self.virtual_device_context]):
+            raise ValidationError(
+                "A VRF must be associated with a device, a virtual machine, or a virtual device context."
+            )
```

Comment:
I feel like these error messages are a bit unclear/misleading - e.g. a VRF **can** be associated with both a device and a virtual-device-context, it just requires two separate entries in the through table. Similarly a VRF does not **have** to be associated with a device, VM, or VDC, it's just that if you're adding an entry in the through table, it needs one of those.